src/Services/UserRegisterService.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core\Services;
+
+use Blumilk\Meetup\Core\Models\User;
+use Illuminate\Auth\Events\Registered;
+use Illuminate\Contracts\Hashing\Hasher;
+
+class UserRegisterService
+{
+    public function __construct(
+        protected Hasher $hasher,
+    ) {}
+
+    public function register(string $email, string $name, string $password): void
+    {
+        if (!$this->hasher->info($password)["algo"]) {
+            $password = $this->hasher->make($password);
+        }
+
+        $user = User::query()->firstOrCreate([
+            "email" => $email,
+            "name" => $name,
+            "password" => $password,
+        ]);
+
+        event(new Registered($user));
```

Comment:
I would return created user in method like that. This is just a comment, you do not need to follow that.

Reply:
I could return the user, but now the controller is one line shorter